components/provider_ppucd/src/structs.rs

```diff
@@ -0,0 +1,92 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/master/LICENSE ).
+
+use icu_uniset::UnicodeSet;
+use serde::{Deserialize, Serialize};
+use std::convert::TryInto;
+
+//
+// data key structs - the structs used directly by users of data provider
+//
+
+macro_rules! data_key {
+    (uniset, $sub_category:literal, $version:tt) => {
+        data_key!(icu_provider::DataCategory::Uniset, $sub_category, $version)
+    };
+
+    // this was copied over from `provider` crate, but maybe should be refactored
+    ($category:expr, $sub_category:literal, $version:tt) => {
+        icu_provider::DataKey {
+            category: $category,
+            sub_category: tinystr::tinystr16!($sub_category),
+            version: $version,
+        }
+    };
+}
+
+pub mod key {
+    use icu_provider::DataKey;
+    pub const WSPACE_V1: DataKey = data_key!(uniset, "WSpace", 1);
+}
+
+/// Gets a locale-invariant default struct given a data key in this module's category.
+#[cfg(feature = "invariant")]
```

Comment:
feature invariant is obsolete now.  Just implement the Default trait on your struct.

Reply:
When I remove this invariant feature attribute, and derive `Default` on `UnicodeProperty`, the compiler still complains that it can't find `DataKey` and `DataResponse` for the function here. Perhaps I need to rebase to pick up the latest changes to data provider? This branch hasn't been rebased for a long time, so I suspect that I will need to rebase either way.

Reply:
You need to get the latest changes from master.

Reply:
Done.